core/iwasm/common/wasm_suspend_flags.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2023 Amazon Inc.  All rights reserved.
```

Comment:
* most of the logic in this file seems not specific to suspend flags. how about using a bit more generic name?
* i guess the default implementation should be c11 atomics.

Reply:
Yes, initially I've had an implementation based on stdatomics, but there are two problems with that:
* The size of the atomic variable is not guaranteed to be the size of the base type - and this is a requirement for JIT. We could potentially workaround it with set of ifs on code generation phase though
* this file is included (indirectly) by some c++ files (fast jit) and removing the dependency would require major refactoring. C11 stdatomics are not available in c++.

For those reasons I decided to go for a simple solution for now, but happy to iterate on that.

Reply:
how about "most of the logic in this file seems not specific to suspend flags. how about using a bit more generic name?"

Reply:
The file right now includes the following:
* definitions of the suspend flag values (`WASM_SUSPEND_FLAG_*` definitions)
* `WASMSuspendFlags` data structure
* macros for operating on the suspend flag data structure (`WASM_SUSPEND_FLAGS_*` macros)
* information whether the suspend flag is atomic (`WASM_SUSPEND_FLAGS_IS_ATOMIC` definition)
* macros for locking the flag if it's not atomic (`WASM_SUSPEND_FLAGS_(UN)LOCK` macros)

They all seem pretty related to suspend flags.

Reply:
availability check and wrappers on atomic builtins are generic.

Reply:
https://github.com/bytecodealliance/wasm-micro-runtime/pull/2393